### ISSUE TITLE
[1.7.4] Useless "skip ingesting data with 0 bytes" messages logged 

### ISSUE THREAD

**Issue by LionelCons** [fluent/fluent-bit#3415]
When using Fluent Bit 1.7.4, we get useless "`skip ingesting data with 0 bytes`" messages logged periodically, at least when using multiline parsing.

Here is how to reproduce the problem.

```
$ cat test.log 
line1 foo=1
  bar=2
line2 foo=2

$ cat test.conf
[SERVICE]
  Flush             1
  Daemon            off
  Parsers_File      parsers.conf

[INPUT]
  Name              tail
  Path              test.log
  Read_from_head    true
  Multiline         on
  Parser_Firstline  test-firstline

[OUTPUT]
  Name              stdout
  Match             *

$ cat parsers.conf
[PARSER]
  Name    test-firstline
  Format  regex
  Regex   ^(?<raw>\S.+)$
```

Using 1.7.4:
```
$ .../fluent-bit -c test.conf 
Fluent Bit v1.7.4
[...]
[2021/04/26 11:36:06] [ info] [sp] stream processor started
[2021/04/26 11:36:06] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/04/26 11:36:06] [ info] [input:tail:tail.0] inotify_fs_add(): inode=2059285632 watch_fd=1 name=test.log
[0] tail.0: [1619429766.665464655, {"raw"=>"line1 foo=1
  bar=2"}]
[0] tail.0: [1619429766.665516413, {"raw"=>"line2 foo=2"}]
[2021/04/26 11:36:13] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/04/26 11:36:17] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/04/26 11:36:21] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/04/26 11:36:25] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/04/26 11:36:29] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/04/26 11:36:33] [ info] [input chunk] skip ingesting data with 0 bytes
^C
[2021/04/26 11:36:34] [engine] caught signal (SIGINT)
```

We get this every 4 seconds (default `Multiline_Flush` value).

In contrast, using 1.7.3:
```
$ .../fluent-bit -c test.conf 
Fluent Bit v1.7.3
[...]
[2021/04/26 11:38:32] [ info] [sp] stream processor started
[2021/04/26 11:38:32] [ info] [input:tail:tail.0] inotify_fs_add(): inode=2059285632 watch_fd=1 name=test.log
[0] tail.0: [1619429912.212860401, {"raw"=>"line1 foo=1
  bar=2"}]
[0] tail.0: [1619429912.212926766, {"raw"=>"line2 foo=2"}]
^C
[2021/04/26 11:39:54] [engine] caught signal (SIGINT)
```

FWIW, this message has been added as part of https://github.com/fluent/fluent-bit/commit/38fcabc1adbcc02c5e33ccb2f1510d92453ab92e.

**Comment by JeffLuoo** [comment]
@edsiper I didn't expect this happens this many times... Is this normal?

**Comment by fkoehler** [comment]
I see the same behaviour. It logs since updating to 1.7.4 every few seconds:

```
[2021/05/12 15:53:35] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:53:35] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:53:50] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:53:50] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:54:21] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:54:21] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               
│ [2021/05/12 15:55:35] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:55:35] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:56:19] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:56:19] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:57:33] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:57:33] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               
│ [2021/05/12 15:58:43] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:58:43] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:59:05] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:59:05] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:59:27] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 15:59:27] [ info] [input chunk] skip ingesting data with 0 bytes                                                                                                                                                                                                                               │
│ [2021/05/12 16:01:25] [ info] [input chunk] skip ingesting data with 0 bytes 
```

config used:

```
[SERVICE]
        Flush                     1
        Log_Level                 info
        Daemon                    off
        Parsers_File              parsers.conf
        HTTP_Server               On
        HTTP_Listen               0.0.0.0
        HTTP_Port                 2020
        storage.path              /var/log/flb-storage/
        storage.sync              full
        storage.checksum          on
        storage.backlog.mem_limit 5M
        storage.metrics           on

    [INPUT]
        Name               tail
        Tag                kube.*
        Path               /var/log/containers/*.log
        Parser             docker
        Docker_Mode        On
        Docker_Mode_Parser docker-multiline
        DB                 /var/log/flb_kube.db
        Mem_Buf_Limit      10MB
        Rotate_Wait        20
        Skip_Long_Lines    On
        Refresh_Interval   10
        storage.type       filesystem
```

**Comment by LionelCons** [comment]
FWIW, this problem is still present in 1.7.5.

**Comment by saputkin** [comment]
We have also encountered similar behaviour in 1.7.4 and 1.7.5.

**Comment by edsiper** [comment]
@JeffLuoo did you find any root cause why buf_size might be zero ?

looks related to multiline or docker mode...

**Comment by kartikzheng** [comment]
We have also encountered similar behaviour in 1.7.8.

**Comment by yukihiko-shinoda** [comment]
We have also encountered similar behavior in 1.7.8.

**Comment by edsiper** [comment]
we changed that message from info -> debug, in the latest versions it's fixed.

**Comment by edsiper** [comment]
Fixed in v1.8, closing the ticket.
